16_wordle/main.go

```diff
@@ -99,3 +99,46 @@ func checkChar(chosenWordSlice []string, inputSlice []string) []string {
 	}
 	return containedSlice
 }
+
+func checkCharPrecise(chosenWordSlice []string, inputSlice []string) []string {
```

Comment:
I really recommend using `[]int` instead, you will get confused later when it comes to compare the values inside the result.

Or, if you want to keep `[]string` for now, make sure that your return is consistent. Currently I list:
- `"Correct word!"`
- `"<char> inplace"`
- `"<char> outofplace"`

Reply:
You almost have the full logic tho:

```go

for inIndex, inChar := range in {
  for chosenIndex, chosenChar := range chosen {
    if inChar != chosenChar{
      // TODO
    } else {
      if inIndex == chosenIndex {
        // TODO
      } else {
        // TODO
      }
    }
  }
}

```

Reply:
Maybe here would be a good idea to start with the tests, and to fit the implementation to the tests?

E.g.
```go

var tests := []struct {
  input string
  chosen string
  want []int
}{
  {
    input: "aaaaa",
    chosen: "aaaab",
    want: []int{2,2,2,2,0},
  },
  {
    input: "acaab",
    chosen: "aacab",
    want: []int{2,1,1,2,2},
  },
  // ...
}

```

Because once you have a list of the different cases, you can implement much faster to fit those cases.